src/main.cpp

```diff
@@ -859,22 +890,51 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransa
         CTxMemPoolEntry entry(tx, nFees, GetTime(), dPriority, chainActive.Height(), mempool.HasNoInputsOf(tx));
         unsigned int nSize = entry.GetTxSize();
 
-        // Don't accept it if it can't get into a block
-        CAmount txMinFee = GetMinRelayFee(tx, nSize, true);
-        if (fLimitFree && nFees < txMinFee)
-            return state.DoS(0, error("AcceptToMemoryPool: not enough fees %s, %d < %d",
-                                      hash.ToString(), nFees, txMinFee),
-                             REJECT_INSUFFICIENTFEE, "insufficient fee");
-
-        // Require that free transactions have sufficient priority to be mined in the next block.
-        if (GetBoolArg("-relaypriority", true) && nFees < ::minRelayTxFee.GetFee(nSize) && !AllowFree(view.GetPriority(tx, chainActive.Height() + 1))) {
-            return state.DoS(0, false, REJECT_INSUFFICIENTFEE, "insufficient priority");
+        // The fees required to accept this transaction start with the fees required to accept it on its own
+        CAmount nFeesRequired = 0;
+        if (fLimitFree) {
+            nFeesRequired = GetMinRelayFee(tx, nSize, true);
+            if (nFees < nFeesRequired)
+                return state.DoS(0, error("AcceptToMemoryPool: not enough fees %s, %d < %d", hash.ToString(), nFees, nFeesRequired),
+                                 REJECT_INSUFFICIENTFEE, "insufficient fee");
+        }
+        // If we are not relaying low priority free transactions, then if this tx doesn't have sufficient priority
+        // it must have minRelayTxFee
+        if (GetBoolArg("-relaypriority", true) && nFeesRequired < ::minRelayTxFee.GetFee(nSize) && !AllowFree(view.GetPriority(tx, chainActive.Height() + 1))) {
+            nFeesRequired =  ::minRelayTxFee.GetFee(nSize);
+            if (nFees < nFeesRequired)
+                return state.DoS(0, false, REJECT_INSUFFICIENTFEE, "insufficient priority");
+        }
+
+        // Try to make space in mempool
+        size_t softcap = GetArg("-maxmempool", DEFAULT_MAX_MEMPOOL_SIZE) * 700000;
+        size_t hardcap = GetArg("-maxmempool", DEFAULT_MAX_MEMPOOL_SIZE) * 1000000;
+        size_t capstep = (hardcap - softcap) / 10;
+        std::set<uint256> stagedelete;
+        CAmount nFeesDeleted = 0;
+
+        if (!mempool.StageTrimToSize(softcap, entry, stagedelete, nFeesRequired, nFeesDeleted)) {
+            size_t expsize = mempool.DynamicMemoryUsage() + mempool.GuessDynamicMemoryUsage(entry);
+            if (expsize > hardcap)
```

Comment:
No objection to changing, but just want to understand what the style request is related to this.   All then blocks should have braces or only because the else block is multi-line?  A single-line then block without braces appears as example code in developer-notes.md.